phpstan.neon.dist

```diff
@@ -5,3 +5,4 @@ parameters:
 	level: max
 	paths:
 		- src
+	checkGenericClassInNonGenericObjectType: false
```

Comment:
tbh I'm more a fan of adding them to the baseline then to deactivated the whole feature for two reasons.

1. This deactivates all checks, not only there where we cant fix it.
2. The baseline can cleanup itself, if we fix it, or phpstan evolves and fixes it itself, since it can understand what we are doing

Reply:
This rule only leads to false positives for us and has not yet caught anything that was correct. We have psalm what Generics understands correctly and has implemented better.

And it's not just a problem that we always have when we do something. But everyone who wants to help develop this library.

Reply:
Well we could add a contribution guideline where we can state out that psalm is for this project first citizen analyser and phpstan not but i get what you mean.

Reply:
I commented on phpstan and see what the feedback is:

https://github.com/phpstan/phpstan/issues/6240#issuecomment-1007364430

Reply:
Hm let's see what the result will be. But until then we would need a decision for this PR.

I would be fine with merging it as it is, since psalm is here our preferred SA tool.